Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonGun.cs

```diff
@@ -248,23 +254,36 @@ private void StartReloadProcess()
         _harpoonFiringState = EHarpoonFiringState.Reloading;
 
         StartCoroutine(ReloadHarpoon());
-
-        RuntimeSfxManager.APlayOneShotSfx?
-            .Invoke(FmodSfxEvents.Instance.HarpoonReload, gameObject.transform.position);
     }
 
     /// <summary>
     /// The process in which the harpoon is reloaded
     /// </summary>
     private IEnumerator ReloadHarpoon()
     {
-        float reloadTimeRemaining = _reloadTime;
-        while(reloadTimeRemaining > 0)
+        while (true)
```

Comment:
If we did have an intermediary function, would it not have basically the same functionality? We need to constantly check to see if we have restocked ammo since running out, and to do that, we would need the function to be another coroutine that uses a while(true) statement.

Reply:
Still disagree, but I'm not dying on this hill. I think avoiding the while(true) would be preferable. To argue with your point of it being the same functionality I don't think so. The while true is checking every frame for if the ammo is empty. With an intermediary function we could have that called only after firing or getting more ammo.

Reply:
I dont see why this needs to be a while(true). Ammo only needs to be checked once before the reload is even called. I say scrap the new functionality here and throw a check for the ammo at the start of the coroutine so it just won't run if theres no ammo.

When ammo is collected, there can be a check to reload the harpoon if it was empty when the ammo was collected